crates/router/src/connector/worldpay/transformers.rs

```diff
@@ -100,9 +128,15 @@ fn fetch_payment_instrument(
     }
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for WorldpayPaymentsRequest {
+impl TryFrom<&WorldpayRouterData<&types::PaymentsAuthorizeRouterData>> for WorldpayPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+
+    fn try_from(item: &WorldpayRouterData<&types::PaymentsAuthorizeRouterData>,) -> Result<Self, Self::Error> {
+        let request = &item.router_data.request;
+        match
+        request.order_details.clone(){
+            Some(order_details)=>
+        
         Ok(Self {
             instruction: Instruction {
                 value: PaymentValue {
```

Comment:
![image](https://github.com/juspay/hyperswitch/assets/105744567/b1fd6d06-3222-4dc7-929f-adbccfffe809)


I think this will solve the issue.
I will commit the changes said.

Reply:
> The `amount` field needs to be retrieved from the `WorldpayRouterData`

@srujanchikke please check and inform if any further changes to be made.

Reply:
@Suraj3240 Why do we need to match with `order_details` ,As We are not using `order details` anywhere in payment request ?


Please refer to this pr for more info  https://github.com/juspay/hyperswitch/pull/2192

Reply:
> @Suraj3240 Why do we need to match with `order_details` ,As We are not using `order details` anywhere in payment request ?
> 
> Please refer to this pr for more info #2192

![image](https://github.com/juspay/hyperswitch/assets/105744567/a877db10-facb-424a-8b23-be17e46b180f)


So I will have to code for all the payment methods like wallet,cards,etc. as given in this file, right?

![image](https://github.com/juspay/hyperswitch/assets/105744567/572c15e8-8954-4319-8966-f28cf00d2b61)
And the amount will be like this, right?

Reply:
You don't have to worry about the other payment methods mapping . But yes , as you mentioned amount should be mapped that way .

Reply:
> @Suraj3240 Why do we need to match with `order_details` ,As We are not using `order details` anywhere in payment request ?
> 
> Please refer to this pr for more info #2192

In this PR [link](https://github.com/juspay/hyperswitch/pull/2414) the fields are derived from the `Order_details`. And hence I used that method.
![image](https://github.com/juspay/hyperswitch/assets/105744567/a764753f-4892-4e66-87c5-ab428605729f)

Reply:
> @Suraj3240 Why do we need to match with `order_details` ,As We are not using `order details` anywhere in payment request ?
> 
> Please refer to this pr for more info #2192

And following the method used in this PR, it is giving out errors.
![image](https://github.com/juspay/hyperswitch/assets/105744567/83597e9f-669a-47c6-a8f3-22d05e1507c8)


So pls someone kindly help me.